data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/LogRateLimiterTest.java

```diff
@@ -0,0 +1,58 @@
+package org.opensearch.dataprepper.plugins.kafka.util;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(MockitoExtension.class)
+public class LogRateLimiterTest {
+
+    @Test
+    public void testRateLimiter() {
+        long currentMs = System.currentTimeMillis();
+        LogRateLimiter objUnderTest = new LogRateLimiter(10, currentMs);
+        assertThat(objUnderTest.isAllowed(currentMs), equalTo(true));
+        assertThat(objUnderTest.isAllowed(currentMs), equalTo(false));
+        currentMs += 50;
```

Comment:
Thank you for manipulating instead of sleeping!

I've made a number of test changes to reduce sleeps that exist in our tests and slow down the build.